### PR TITLE
SALTO-6130 Pagerduty fix layers with users

### PR DESCRIPTION
**Pull request by Uri-Bar**
_Replace me with a description of the changes in this PR_

---

_Additional context for reviewer_

---
_Release Notes_: 
_Replace me with a short sentence that describes the effect of this change on Salto users_

---
_User Notifications_: 
_Replace me with a short sentence that describes changes that will appear in NaCls and are not caused by user actions (e.g. a new annotation, field values that are converted to references, etc). Hidden changes should not be listed._

### CODE REVIEW

File: packages/pagerduty-adapter/src/config.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export type UserConfig = definitions.UserConfig<
   definitions.UserDeployConfig
 >
 
-export const DEFAULT_CONVERT_USERS_IDS_VALUE = false
+export const DEFAULT_CONVERT_USERS_IDS_VALUE = true
```
Comment (edenhassid) [inline packages/pagerduty-adapter/src/config.ts:32]: why did you change this?

File: packages/pagerduty-adapter/src/definitions/utils/schedule_layers.ts [hunk 2]
```diff
@@ -17,10 +17,29 @@ import _ from 'lodash'
 import { definitions } from '@salto-io/adapter-components'
 import { getChangeData } from '@salto-io/adapter-api'
 import { getParent } from '@salto-io/adapter-utils'
+import { values as lowerdashValues } from '@salto-io/lowerdash'
+
+export const addStartTime = (value: unknown): Record<string, unknown> => {
```
Comment (edenhassid) [inline packages/pagerduty-adapter/src/definitions/utils/schedule_layers.ts:22]: should this be exported?

File: packages/pagerduty-adapter/src/definitions/utils/schedule_layers.ts [hunk 3]
```diff
@@ -17,10 +17,29 @@ import _ from 'lodash'
...
+  }
+  return { ...value, start: value.rotation_virtual_start }
+}
+
+export const addStartToLayers: definitions.AdjustFunction<definitions.deploy.ChangeAndContext> = ({ value }) => {
+  if (!lowerdashValues.isPlainRecord(value)) {
+    throw new Error('Can not adjust when the value is not an object')
+  }
+  const layers = _.get(value, 'schedule.schedule_layers')
+  if (!Array.isArray(layers)) {
```
Comment (edenhassid) [inline packages/pagerduty-adapter/src/definitions/utils/schedule_layers.ts:34]: can they be undefined?

File: packages/pagerduty-adapter/src/definitions/utils/schedule_layers.ts [hunk 4]
```diff
@@ -17,10 +17,29 @@ import _ from 'lodash'
 import { definitions } from '@salto-io/adapter-components'
 import { getChangeData } from '@salto-io/adapter-api'
 import { getParent } from '@salto-io/adapter-utils'
+import { values as lowerdashValues } from '@salto-io/lowerdash'
+
+export const addStartTime = (value: unknown): Record<string, unknown> => {
+  if (!lowerdashValues.isPlainRecord(value)) {
+    throw new Error('Can not adjust when the value is not an object')
+  }
+  return { ...value, start: value.rotation_virtual_start }
```
Comment (edenhassid) [inline packages/pagerduty-adapter/src/definitions/utils/schedule_layers.ts:26]: I think you should check that `value.rotation_virtual_start` is not undefined

File: packages/pagerduty-adapter/src/filters/users.ts [hunk 5]
```diff
@@ -95,14 +104,14 @@ const replaceValues = (instance: InstanceElement, mapping: Record<string, string
...
-            _.set(userObj, 'user.id', mapping[userIdentifier])
-          } else {
-            log.debug(`Could not find user with id ${userIdentifier} in the mapping`)
-          }
-        })
+        .forEach(userObj => replaceUserInUserObject(userObj, mapping))
+      break
+    case SCHEDULE_TYPE_NAME:
+      makeArray(instance.value.schedule_layers).forEach(scheduleLayer => {
+        makeArray(scheduleLayer.resValue.value.users)
```
Comment (edenhassid) [inline packages/pagerduty-adapter/src/filters/users.ts:111]: you shouldn't change things inside a reference expression. also it can be an unresolved reference and you will get an error here. you should resolve the instance before this filter or in this filter and then the list of schedule layers will not be references.

**Review by edenhassid — COMMENTED** [review]
looks good! 💯  added some comments. don't forget to fill in the release notes

File: packages/pagerduty-adapter/src/definitions/utils/schedule_layers.ts (hunk 2 above)
Comment (Uri-Bar) [reply packages/pagerduty-adapter/src/definitions/utils/schedule_layers.ts:22]: will remove

File: packages/pagerduty-adapter/src/config.ts (hunk 1 above)
Comment (Uri-Bar) [reply packages/pagerduty-adapter/src/config.ts:32]: f2f

File: packages/pagerduty-adapter/src/definitions/utils/schedule_layers.ts (hunk 3 above)
Comment (Uri-Bar) [reply packages/pagerduty-adapter/src/definitions/utils/schedule_layers.ts:34]: fixed, will add log

File: packages/pagerduty-adapter/src/definitions/utils/schedule_layers.ts (hunk 4 above)
Comment (Uri-Bar) [reply packages/pagerduty-adapter/src/definitions/utils/schedule_layers.ts:26]: fixed

File: packages/pagerduty-adapter/src/filters/users.ts (hunk 5 above)
Comment (Uri-Bar) [reply packages/pagerduty-adapter/src/filters/users.ts]: fixed

**Review by edenhassid — APPROVED** [review]
LGTM! 🔥 🏆
